### PR TITLE
SBAELIDEV-13 Integrate SBAELIDEV-205 with for dynamic search result experience

### PR DESCRIPTION
**Pull request by kdyer**

### CODE REVIEW

File: service-layer/app/models/concerns/queryable.rb [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+# to ActiveRecords `find_by_sql` method. For example, lets say you have a query
+# that you want to union together results from two different tables.
+#
+#   SELECT id, name, 'learning_path' AS content_type FROM learning_paths
+#   UNION ALL
+#   SELECT if, name, 'learning_objective' AS content_type FROM learning_objectives
+#
+# For this query, you would not want to use the ActiveRecord find_by_sql method
+# because the result set would contain more than just one model type. Queryable
+# addressese this by letting you create a generic Ruby class and including the
```
Comment (arwagner) [inline service-layer/app/models/concerns/queryable.rb]: Typo: "addresses"

File: service-layer/app/models/concerns/queryable.rb [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+          count_query = original_query.sub /\bORDER\s+BY\s+[\w`,\s.]+$/mi, ""
+          count_query = "SELECT COUNT(*) FROM (#{count_query})"
+          count_query << " AS count_table"
+          # perform the count query
+          pager.total_entries = count_by_sql(count_query)
+        end
+      end
+    end
+
+    # Convenience method for sanitizin a SQL Statement
```
Comment (arwagner) [inline service-layer/app/models/concerns/queryable.rb]: Typo: "sanitizin"

File: service-layer/app/models/learning_event.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#
+#  fk_rails_...  (cloned_item_id => cloned_items.id)
+#  fk_rails_...  (context_module_id => context_modules.id)
+#  fk_rails_...  (learning_outcome_id => learning_outcomes.id)
+#
+
+class LearningEvent < ApplicationRecord
+  include Contentable
+
+  self.table_name = "content_tags"
```
Comment (arwagner) [inline service-layer/app/models/learning_event.rb]: I'm not sure I understand this. Why would the table name for LearningEvent be content_tags? I assume that's not a Canvas table, but our own new one. Might there be custom data that we store here other than tags?

File: service-layer/app/models/learning_path.rb [hunk 4]
```diff
@@ -81,7 +81,12 @@
...
 
 class LearningPath < ApplicationRecord
+  include Contentable
+
   self.table_name = "courses"
 
   acts_as_taggable_on :subjects
+  acts_as_taggable_on :duration
+
+  serialize :settings
```
Comment (arwagner) [inline service-layer/app/models/learning_path.rb]: What is this settings field used for?

File: service-layer/app/models/concerns/queryable.rb (hunk 2 above)
Comment (nwwatson) [reply service-layer/app/models/concerns/queryable.rb]: Fixed in the correct pull request: https://github.com/FearlessSolutions/sba-eli/pull/61/commits/34828d3571513f736e2dde1aafbc8aa62287e660

File: service-layer/app/models/concerns/queryable.rb (hunk 1 above)
Comment (nwwatson) [reply service-layer/app/models/concerns/queryable.rb]: Fixed in the correct pull request: https://github.com/FearlessSolutions/sba-eli/pull/61/commits/34828d3571513f736e2dde1aafbc8aa62287e660

**Pull request merged**